common/archcheck.cc

```diff
@@ -136,11 +138,25 @@ void archcheck_conn(const Context *ctx)
         for (PipId pip : ctx->getPipsDownhill(wire)) {
             WireId wire2 = ctx->getPipSrcWire(pip);
             log_assert(wire == wire2);
+
+            auto iter = pips_downhill.find(pip);
+            if(iter != pips_downhill.end()) {
+                log_assert(iter->second == wire);
+            } else {
+                pips_downhill.insert(std::make_pair(pip, wire));
```

Comment:
unless there's a subtle difference I'm missing, I think `emplace` is neater than `insert` and `make_pair`?

Reply:
Fixed.